### ISSUE TITLE
Prediction on NER Tensorflow 2

### ISSUE THREAD

**Issue by imayachita** [huggingface/transformers#2559]
Hi,
I tried running the implementation of NER on Tensorflow 2. I have a problem doing the prediction. Seems like the label to index are off. Here is some examples:

```
SOCCER B-ORG
- B-ORG
JAPAN B-MISC
GET B-ORG
LUCKY B-MISC
WIN B-ORG
, B-ORG
CHINA B-MISC
IN B-ORG
SURPRISE O
DEFEAT B-ORG
. B-ORG

Nadim B-MISC
Ladki B-PER

AL-AIN I-PER
, B-ORG
United I-PER
Arab I-MISC
Emirates I-MISC
1996-12-06 B-ORG

Japan I-PER
began B-ORG
the B-ORG
defence B-ORG
of B-ORG
their B-ORG
Asian O
Cup I-ORG
title B-ORG
with B-ORG
a B-ORG
lucky B-ORG
2-1 B-ORG
win B-ORG
against B-ORG
Syria I-PER
in B-ORG
a B-ORG
Group B-ORG
C I-ORG
championship B-ORG
match B-ORG
on B-ORG
Friday B-ORG
. B-ORG
```

Anyone has the same problem? Thanks!

**Comment by HuiyingLi** [comment]
Yes, the output is wrong. 
I think the run_tf_ner.py script has a bug where the labels are off by 1. 
And the off-by-1 prediction result is sent for evaluation `metrics.classification_report(y_true, y_pred, digits=4)` therefore the evaluation result is wrong too.

**Comment by imayachita** [comment]
Thanks @HuiyingLi! The workaround works.

**Comment by andrey999333** [comment]
I'm not 100% sure, but it looks like there is a bug in the implementation of the training. The loss function 
`loss_fct = tf.keras.losses.SparseCategoricalCrossentropy(reduction=tf.keras.losses.Reduction.NONE)` expects probabilities but bert models, as they are in transformers, return logits. So additional argument `from_logits=True` should be passed to the `tf.keras.losses.SparseCategoricalCrossentropy`, coz default is `False`
